shared/src/messages.cpp

```diff
@@ -0,0 +1,37 @@
+#include "messages.hpp"
+#include <iostream>
+#include <stdexcept>
+
+std::string format_message(const msg_t &msg) {
+    std::string str_msg("AIRv1.0 " + msg.receiver_id + " SM\n" + msg.body + "\nEM " + msg.caller_id);
+    return str_msg;
+}
+
+msg_t parse_message(const std::string &str_msg) {
+
+    const size_t start_msg = str_msg.find("SM");
+    const size_t end_msg = str_msg.find("EM");
+    const std::string msg_header = "AIRv1.0";
```

Comment:
Since this is repeated, instead make a variable at the top of the class that is
`static constexpr std::string header = "AIRv1.0"`